### ISSUE TITLE
[fvt] SN will dispatch  "nodeset <node> offline" command back to MN 

### ISSUE THREAD

**Issue by hu-weihua** [xcat2/xcat-core#3930]
In  hierarchy environment,  SN dispatch  "nodeset <node> offline" command back to MN.

```
[root@briggs01 ~]# lsdef mid05tor12cn11
Object name: mid05tor12cn11
    arch=ppc64le
    bmc=172.12.139.111
    chain=runcmd=bmcsetup,standby
    cons=openbmc
    conserver=sn02
    cpucount=144
    cputype=POWER9 (raw), altivec supported
    currchain=standby
    currstate=offline
    disksize=sda:1000GB,sdb:1000GB
    groups=all,compute,cn_regex
    ip=172.12.139.11
    mac=70:e2:84:14:29:10
    memory=523491MB
    mgt=openbmc
    monserver=sn02
    mtm=8335-GTC
    netboot=petitboot
    nodetype=mp
    os=rhels7.4-snap2
    postbootscripts=otherpkgs
    postscripts=syslog,remoteshell,syncfiles
    profile=compute
    provmethod=rhels7.4-snap2-ppc64le-netboot-compute_mlnx_cuda
    serial=13189CA
    servicenode=sn02
    status=netbooting
    statustime=09-14-2017 06:57:46
    supportedarchs=ppc64
    switch=mid05tor12
    switchport=11
    xcatmaster=172.12.254.2

[root@briggs01 ~]# nodeset mid05tor12cn11 offline -V
mid05tor12cn11: [briggs01]: offline
mid05tor12cn11: [sn02]: offline
Error: [sn02]: Unable to dispatch hierarchical sub-command to 172.11.253.27:3001.  Error: Connection failure: IO::Socket::INET: connect: Network is unreachable at /opt/xcat/lib/perl/xCAT/Client.pm line 248.
.
```
The ip 172.11.253.27 is one ip address of MN.
sn02 should not dispatch this command back to MN. 
Another interesting thing is `` 172.11.0.0`` is hardware control network between MN and SN, why did sn02 choose this network to dispatch command?

**Comment by robin2008** [comment]
There is two issues:

1. 
In the petitboot plugin,  the `offline` action to handle 'makedhcp' has some issue,  it should not dispatch the request to others if site.sharedtftp=no.

2. 
For dhcp dispatching, it is required to filter the non-accessing IP from the dispatching list.

**Comment by robin2008** [comment]
Fix it by PR #3954
